data/mods/ssb/moves.ts

```diff
@@ -667,6 +667,94 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Fire",
 	},
 
+	// Celestial
+	pandorasbox: {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "Please refer to the guide for details on this move.",
+		shortDesc: "Please refer to the guide for details on this move.",
+		name: "Pandora's Box",
+		pp: 8,
+		priority: 1,
+		flags: {},
+		volatileStatus: 'pandorasbox',
+		condition: {
+			onStart(target) {
+				const typeMovePair: {[key: string]: string} = {
+					'Normal': 'Body Slam',
+					'Fighting': 'Drain Punch',
+					'Flying': 'Floaty Fall',
+					'Poison': 'Baneful Banker',
+					'Ground': 'Shore Up',
+					'Rock': 'Stealth Rock',
+					'Bug': 'Sticky Web',
+					'Ghost': 'Shadow Sneak',
+					'Steel': 'Iron Defense',
+					'Fire': 'Fire Fang',
+					'Water': 'Life Dew',
+					'Grass': 'Synthesis',
+					'Electric': 'Thunder Fang',
+					'Psychic': 'Psychic Fangs',
+					'Ice': 'Icicle Crash',
+					'Dragon': 'Dragon Darts',
+					'Dark': 'Taunt',
+					'Fairy': 'Play Rough',
+				};
+				const foe = target.side.foe;
+				const types = [foe.pokemon[this.random(foe.pokemon.length)].types[0]];
+				for (const pokemon of foe.pokemon) {
+					if (!types.includes(pokemon.types[0])) {
+						types.push(pokemon.types[0]);
+						break;
+					}
+				}
+				const moves = [typeMovePair[types[0]], typeMovePair[types[1]]];
+				target.m.replacedMoves = [typeMovePair[types[0]], typeMovePair[types[1]]];
```

Comment:
if i do that, ```moves.pop()``` will also pop ```target.m.replacedMoves``` since when referring to data types like array or objects, their address is used instead of their value i think

Reply:
Yeah I figured that was something to do with how JS objects worked, just checking.

Reply:
theres no need to change this then